src/containerapp/azext_containerapp/_cloud_build_utils.py

```diff
@@ -130,27 +130,32 @@ def spin():
 
         # Source code compression
         done_spinner = False
-        thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
-        tar_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
-        archive_source_code(tar_file_path, source)
+        source_file_path = source
+        if os.path.isdir(source):
+            thread = display_spinner(f"Compressing data: {font_bold}{source}{font_default}")
+            source_file_path = os.path.join(tempfile.gettempdir(), f"{build_name}.tar.gz")
+            archive_source_code(source_file_path, source)
         done_spinner = True
         thread.join()
 
         # File upload
         done_spinner = False
-        thread = display_spinner("Uploading compressed data")
+        thread = display_spinner("Uploading data")
         headers = {'Authorization': 'Bearer ' + token}
         try:
-            tar_file = open(tar_file_path, "rb")
-            files = [("file", ("build_data.tar.gz", tar_file, "application/x-tar"))]
+            source_file = open(source_file_path, "rb")
+            file_extension = os.path.splitext(source_file_path)[1]
+            files = [("file", ("build_data" + file_extension, source_file))]
```

Comment:
I see that before we were marking specifically the type as "application/x-tar". Could you please confirm how the behavior works when this isn't set? Is `requests` able to deduct that value from the file extension?

Reply:
Hi @daniv-msft 
The content type is never used in RP, only file extension is used to determine if the file type is supported:
https://msazure.visualstudio.com/DefaultCollection/One/_git/AAPT-Antares-Containerapps?path=/src/Microsoft.ContainerApps/Microsoft.ContainerApps.ProxyApi/Controllers/BuildApiController.cs&version=GBmain&line=53&lineEnd=53&lineStartColumn=3&lineEndColumn=96&lineStyle=plain&_a=contents